src/components/elements/Map-mapbox/utils.ts

```diff
@@ -375,29 +372,82 @@ export const addSourceToLayer = (layer: any, map: mapboxgl.Map, polygonsData: Re
       map.removeSource(name);
     }
     const URL_GEOSERVER = `${GEOSERVER}/geoserver/gwc/service/wmts?REQUEST=GetTile&SERVICE=WMTS
-      &VERSION=1.0.0&LAYER=${WORKSPACE}:${name}&STYLE=&TILEMATRIX=EPSG:900913:{z}&TILEMATRIXSET=EPSG:900913&FORMAT=application/vnd.mapbox-vector-tile&TILECOL={x}&TILEROW={y}&RND=${Math.random()}`;
+      &VERSION=1.0.0&LAYER=${WORKSPACE}:${layerName}&STYLE=&TILEMATRIX=EPSG:900913:{z}&TILEMATRIXSET=EPSG:900913&FORMAT=application/vnd.mapbox-vector-tile&TILECOL={x}&TILEROW={y}&RND=${Math.random()}`;
     map.addSource(name, {
       type: "vector",
       tiles: [URL_GEOSERVER]
     });
     styles?.forEach((style: LayerWithStyle, index: number) => {
-      addLayerStyle(map, name, style, index);
+      addLayerStyle(map, name, layerName, style, index);
     });
-    loadLayersInMap(map, polygonsData);
+    loadLayersInMap(map, polygonsData, layer);
   }
 };
-
-export const addLayerStyle = (map: mapboxgl.Map, sourceName: string, style: LayerWithStyle, index: number) => {
+const loadDeleteLayer = (layer: any, map: mapboxgl.Map, polygonsData: Record<string, string[]> | undefined) => {
+  const { name, layerName, styles } = layer;
+  styles?.forEach((style: LayerWithStyle, index: number) => {
+    if (map.getLayer(`${name}-${index}`)) {
+      map.removeLayer(`${name}-${index}`);
+    }
+    map.addLayer({
+      ...style,
+      id: `${name}-${index}`,
+      source: name,
+      "source-layer": layerName
+    } as mapboxgl.AnyLayer);
+  });
+  loadLayersInMap(map, polygonsData, layer);
+};
+export const addDeleteLayer = (layer: any, map: mapboxgl.Map, polygonsData: Record<string, string[]> | undefined) => {
+  const { name, layerName, styles } = layer;
+  if (map) {
+    if (map.getSource(name)) {
+      styles?.forEach((_: unknown, index: number) => {
+        map.removeLayer(`${name}-${index}`);
+      });
+      map.removeSource(name);
+    }
+    const URL_GEOSERVER = `${GEOSERVER}/geoserver/gwc/service/wmts?REQUEST=GetTile&SERVICE=WMTS
```

Comment:
I guess coming back to the naming stuff, this is not the GEOSERVER_URL, it is about the tile, right?
can rename this?

Reply:
changed to